src/collection.ts

```diff
@@ -305,7 +348,11 @@ class Collection {
    * @param {Collection~insertOneWriteOpCallback} [callback] The command result callback
    * @returns {Promise<void>} returns Promise if no callback passed
    */
-  insertOne(doc: object, options?: any, callback?: Function): Promise<void> {
+  insertOne(
+    doc: Document,
+    options?: InsertOperationOptions,
```

Comment:
`InsertOneOptions`, even if it ends up being the same I think we should have parity (with our other methods and with the CRUD spec)